### PR TITLE
Sparrow dom/messaging requirement

### PR DESCRIPTION
**Pull request by sparrowDom**
### Description:
- Re-introduce messaging as part of on-boarding process 
- prefer `useQuery` and `useState` hooks instead of components
- show Rewards and Messaging onboarding part only once. If a user skips it do not show it again
- add Puppeteer tests
- clean up `messaging-client` signing messages code to be able to catch errors
- show errors on failed signing of messaging string and offer actions to restart / skip process

**Solves issues:** https://github.com/OriginProtocol/origin/issues/2936 https://github.com/OriginProtocol/origin/issues/1940 https://github.com/OriginProtocol/origin/issues/1937
### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [x] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [x] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- ~[ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)~

### CODE REVIEW

**Comment by sparrowDom** [comment]
@nick @shahthepro just pinging so this isn't overlooked ☝️

File: dapps/marketplace/src/pages/onboard/Messaging.js [hunk 1]
```diff
@@ -57,44 +58,35 @@ const MessagingSyncing = ({ pct }) => (
...
-      <fbt desc="onboard.Messaging.zeroOfTwoSigned">
-        {' '}
-        0 of 2 MetaMask messages signed
-      </fbt>
-    </div>
+const EnableMessaging = ({ firstMessageSigned, buttons, error }) => {
+  return (
+    <div className="onboard-box">
+      <div className="messaging-logo" />
+      <div className="status">Origin Messaging</div>
```
Comment (shahthepro) [inline dapps/marketplace/src/pages/onboard/Messaging.js]: Translation is missing here

Comment (sparrowDom) [reply dapps/marketplace/src/pages/onboard/Messaging.js]: nice catch thanks!

File: dapps/marketplace/src/pages/App.js [hunk 2]
```diff
@@ -44,7 +44,11 @@ class App extends Component {
     hasError: false,
     displayMobileModal: false,
     mobileModalDismissed: false,
-    footer: false
+    footer: false,
+    onboard: {
+      skipMessaging: false,
```
[the remark this answers is not in the document]
Comment (sparrowDom) [reply dapps/marketplace/src/pages/App.js]: yeah couldn't decide what approach would be better. Will let myself be nudged :)

**Pull request merged**